### PR TITLE
Fix hot module replacement 

### LINKED ISSUE #387: javascript hot code reloading not working in demo app
The hot code reloading does not work if:

* code outside of the ui frontend part, e.g. the libraries, is changed
* scss is changed
* files are added to the ui frontend

### PR DESCRIPTION
**Pull request by chrismatix**
resolves #387

### CODE REVIEW

File: frontend/ui/src/reducers/data/conversations/index.ts [hunk 1]
```diff
@@ -207,10 +207,10 @@ function allReducer(
             ...state.items[action.payload.identifier],
             metadata: {
               // Ensure that there is always a display name present
-              ...(<MetadataEvent<ConversationMetadata>>action.payload).metadata,
```
Comment (chrismatix) [inline frontend/ui/src/reducers/data/conversations/index.ts:210]: This way of doing type assertions is outdated and conflicts with the babel typescript plugin

**Review by lucapette — DISMISSED** [review]

File: lib/typescript/httpclient/client.ts [hunk 2]
```diff
@@ -47,7 +77,7 @@ export class HttpClient {
       try {
         return await response.json();
       } catch {
-        // NOP
+        // NOOP
```
Comment (AitorAlgorta) [inline lib/typescript/httpclient/client.ts:80]: 💣

**Review by AitorAlgorta — APPROVED** [review]
Good stuff

**Comment by chrismatix** [comment]
@bitboxer please come back. See I fixed it 🙇

**Comment by bitboxer** [comment]
I am now working on [recover](https://www.recoverapp.de/) but will happily switch back to airy if you can make your boss pay me 😉 .

**Comment by steffh** [comment]
i always expected you to come back one day, but this was kind of fast. 😉

**Pull request merged**